### PR TITLE
net/ftp-proxy: Fix php errors in ServiceController

### PR DESCRIPTION
**Pull request by Monviech**
Fixes: https://github.com/opnsense/plugins/issues/4129

Added checks to ensure $this->request is set before accessing it.

Error: Typed property OPNsense\Mvc\Controller::$request must not be accessed before initialization in /usr/local/opnsense/mvc/app/controllers/OPNsense/FtpProxy/Api/ServiceController.php:52

### CODE REVIEW

**Comment by fichtner** [comment]
Feel free to drop the conditionals completely because `sessionClose()` is a no-op now.

File: net/ftp-proxy/src/opnsense/mvc/app/controllers/OPNsense/FtpProxy/Api/ServiceController.php [hunk 1]
```diff
@@ -62,7 +59,7 @@ public function statusAction($uuid)
     public function startAction($uuid)
     {
         $result = array("result" => "failed", "function" => "start");
-        if ($this->request->isPost()) {
+        if (isset($this->request) && $this->request->isPost()) {
             $this->sessionClose();
         }
```
Comment (fichtner) [inline net/ftp-proxy/src/opnsense/mvc/app/controllers/OPNsense/FtpProxy/Api/ServiceController.php:64]:
```suggestion
```

Comment (Monviech) [reply net/ftp-proxy/src/opnsense/mvc/app/controllers/OPNsense/FtpProxy/Api/ServiceController.php]: Sheesh you were faster than me. I just got yelled at by git in the command line xD

**Comment by fichtner** [comment]
Merged, thanks!

**Comment by Madifor1972** [comment]
When can an updated plugin being expected for opnsense??
Current plugin version : 1.0_3

**Comment by fichtner** [comment]
Just test it yourself. This came in only 1 hour ago so it will take a number of days to end up in a release at the very least.

    # opnsense-patch -c plugins f611658

**Comment by Madifor1972** [comment]
Thank you for this information, this really helps me and saved my day :)

great work

Op di 30 jul 2024 om 08:01 schreef ***@***.*** <
***@***.***>:

**Pull request merged**